### PR TITLE
Fix issue #209

### PR DESCRIPTION
**Pull request by DarrenWestwood**
Fixes issue #209 
Uses the new post meta titles to display the transaction info in the woocommerce order and email.

### CODE REVIEW

File: blockonomics-woocommerce.php [hunk 1]
```diff
@@ -343,20 +343,30 @@ function show_basic() {
...
         include_once plugin_dir_path(__FILE__) . 'php' . DIRECTORY_SEPARATOR . 'Blockonomics.php';
-        if ($txid && $address) {
-            echo '<h2>'.__('Payment Details', 'blockonomics-bitcoin-payments').'</h2><p><strong>'.__('Transaction', 'blockonomics-bitcoin-payments').':</strong>  <a href =\''. Blockonomics::BASE_URL ."/api/tx?txid=$txid&addr=$address'>".substr($txid, 0, 10). '</a></p><p>'.__('Your order will be processed on confirmation of above transaction by the bitcoin network.', 'blockonomics-bitcoin-payments').'</p>';
+        if ($btc_txid && $btc_address) {
+            echo '<h2>'.__('Payment Details', 'blockonomics-bitcoin-payments').'</h2><p><strong>'.__('Transaction', 'blockonomics-bitcoin-payments').':</strong>  <a href =\''. Blockonomics::BASE_URL ."/api/tx?txid=$btc_txid&addr=$btc_address'>".substr($btc_txid, 0, 10). '</a></p><p>'.__('Your order will be processed on confirmation of above transaction by the bitcoin network.', 'blockonomics-bitcoin-payments').'</p>';
+        }
+        $bch_txid = get_post_meta($order->get_id(), 'blockonomics_bch_txid', true);
+        $bch_address = get_post_meta($order->get_id(), 'bch_address', true);
```
Comment (shivaenigma) [inline blockonomics-woocommerce.php:353]: Code duplication and should be avoided. Use *txid* and *address* variable. *url* can also be a variable. It is unlikely that customer get both bch and btc tx in one order (this case can be ignored)

File: blockonomics-woocommerce.php [hunk 2]
```diff
@@ -343,20 +343,30 @@ function show_basic() {
...
+        $btc_address = get_post_meta($order->get_id(), 'btc_address', true);
+        include_once plugin_dir_path(__FILE__) . 'php' . DIRECTORY_SEPARATOR . 'Blockonomics.php';
+        if ($btc_txid && $btc_address) {
+          echo '<h2>Payment Details</h2><p><strong>'.__('Transaction').':</strong>  <a href =\''. Blockonomics::BASE_URL ."/api/tx?txid=$btc_txid&addr=$btc_address'>".substr($btc_txid, 0, 10). '</a></p>';
+        }
+        $bch_txid = get_post_meta($order->get_id(), 'blockonomics_bch_txid', true);
+        $bch_address = get_post_meta($order->get_id(), 'bch_address', true);
+        if ($bch_txid && $bch_address) {
+          echo '<h2>Payment Details</h2><p><strong>'.__('Transaction').':</strong>  <a href =\''. Blockonomics::BCH_BASE_URL ."/api/tx?txid=$bch_txid&addr=$bch_address'>".substr($bch_txid, 0, 10). '</a></p>';
         }
```
Comment (shivaenigma) [inline blockonomics-woocommerce.php:370]: Avoid code duplication as above

**Comment by DarrenWestwood** [comment]
Removes code duplication and fixes issue not displaying the transaction info before the transaction is confirmed.

File: php/Blockonomics.php [hunk 3]
```diff
@@ -524,8 +525,7 @@ public function check_callback_secret($secret){
     }
 
     // Save the received payment info to the WooCommerce order
-    public function record_payment($value, $order, $wc_order){
-        update_post_meta($wc_order->get_id(), 'paid_'. $order['crypto'], $value/1.0e8);
+    public function record_transaction($value, $order, $wc_order){
```
Comment (shivaenigma) [inline php/Blockonomics.php:528]: The code comment is still referring to the older. Both the names record_transaction adn record_payment are particularly similar, so we should attempt to clarify with correct code comments or more distinct function names

File: php/Blockonomics.php [hunk 4]
```diff
@@ -538,6 +538,19 @@ public function is_payment_recorded($status, $order, $wc_order){
         return true;
     }
 
+    public function is_transaction_recorded($status, $order, $wc_order){
+        if (!metadata_exists('post',$wc_order->get_id(),'blockonomics_'. $order['crypto'] .'_txid', $order['txid']) )  {
+            return false;
+        }
+        return true;
+    }
```
Comment (shivaenigma) [inline php/Blockonomics.php:546]: *is_transaction_recorded* and *record_transaction* can be merged into a single function. 
It is not a particularly good idea to keep having very small functions (unless they are used multiple times) as it leads to unnecessary function calls and code complexity

Comment (DarrenWestwood) [reply php/Blockonomics.php]: Renamed and refactored in 9e0e8b2

**Pull request merged**